### DISCUSSION TITLE
Export/build the nixvim configuration for usage outside of Nix ecosystem

### DISCUSSION THREAD

**Discussion by DataHearth** [nix-community/nixvim#2550] [Q&A] [answered]
Hey there!

I have a quick, but quick annoying and weird given the context of Nixvim ^^. 
My current work doesn't doesn't support nix in the development environment. Like, it's not possible for me to have the Nix store due to high storage storage (less than 5GB for everything, repo and tools). 

Over time, I've build my development workflow, OSs around Nix and NixOS, and I'm not willing to give it up! So I thought, is there a way to export my current Nixvim configuration as plain lua, even the ugliest is OK, as long I can use it at work. It doesn't even need to be up-to-date or easy to update. Just works once, it's ok!

Thanks!

**Comment by MattSturgeon** [comment, marked as answer]
You can use `nixvim-print-init` to view `init.lua`.

Assuming you don't use any `extraFiles`, that will be the entirety of your config.

Going forward it'd be nice to be able to also access all output files via `passthru` attrs and/or option values, but currently that may not be super easy. Certainly not well documented.

But currently we only have `nixvim-print-init`.

If you use that lua file, you'll have to edit anything that references the nix store. 

You'll also need to find another way to install any plugins you use and add them to vim's runtimepath. By default nixvim installs vim plugins to the nix store.

**Reply by DataHearth** [reply]
Awesome! Gonna take a look tomorrow, and probably do a small python script to update paths referencing the store. 

Thanks! I'll possibly take a closer look to check how I would handle a small documentation for a such niche use case 😋.
